source/opt/folding_rules.cpp

```diff
@@ -2507,24 +2507,16 @@ FoldingRule RedundantFMix() {
   };
 }
 
-// This rule handles addition of zero for integers.
-FoldingRule RedundantIAdd() {
-  return [](IRContext* context, Instruction* inst,
+FoldingRule RedundantBinaryLRhs0(spv::Op op, size_t arg) {
```

Comment:
I would not pass in the `op` just to be used in the assert. I would assert that you can find `inst->opcode()` in `RedundantBinaryRhs0Ops`. It is slow, but only for the debug builds.